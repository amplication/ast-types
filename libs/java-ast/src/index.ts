// Core exports
export * from "./lib/core/AstNode";
export * from "./lib/core/Writer";

// AST nodes exports
export * from "./lib/ast/Access";
export * from "./lib/ast/Annotation";
export * from "./lib/ast/Class";
export * from "./lib/ast/ClassReference";
export * from "./lib/ast/CodeBlock";
export * from "./lib/ast/Enum";
export * from "./lib/ast/Field";
export * from "./lib/ast/GenericType";
export * from "./lib/ast/Import";
export * from "./lib/ast/Interface";
export * from "./lib/ast/Method";
export * from "./lib/ast/Parameter";
export * from "./lib/ast/Type";

// Utils exports
// (Add utility functions as needed)
