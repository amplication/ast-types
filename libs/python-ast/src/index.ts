export * from "./lib/ast";
export * from "./lib/core";
