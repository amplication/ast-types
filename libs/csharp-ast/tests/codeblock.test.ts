import { CsharpSupport as csharp } from "../src";

describe("code block", () => {
  it("string codeblock with references", () => {
    const myClass = csharp.class_({
      name: "MyClassName",
      namespace: "Amplication",
      partial: true,
      access: "public",
      abstract: true,
    });

    myClass.addMethod(
      csharp.method({
        access: "public",
        name: "MyVoidMethod",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "simpleParameter",
            type: csharp.Types.integer(),
          }),
          csharp.parameter({
            name: "listParameter",
            type: csharp.Types.list(csharp.Types.string()),
          }),
          csharp.parameter({
            name: "optionalParameter",
            type: csharp.Types.optional(csharp.Types.string()),
          }),
        ],
        body: csharp.codeblock({
          code: 'Debug.WriteLine("fake log");',
          references: [
            csharp.classReference({
              name: "Debug",
              namespace: "System.Diagnostics",
            }),
          ],
        }),
      })
    );

    expect(myClass.name).toBe("MyClassName");
    expect(myClass.toString()).toMatchSnapshot();
  });

  it("string codeblock without references", () => {
    const myClass = csharp.class_({
      name: "MyClassName",
      namespace: "Amplication",
      partial: true,
      access: "public",
      abstract: true,
    });

    myClass.addMethod(
      csharp.method({
        access: "public",
        name: "MyVoidMethod",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "simpleParameter",
            type: csharp.Types.integer(),
          }),
          csharp.parameter({
            name: "listParameter",
            type: csharp.Types.list(csharp.Types.string()),
          }),
          csharp.parameter({
            name: "optionalParameter",
            type: csharp.Types.optional(csharp.Types.string()),
          }),
        ],
        body: csharp.codeblock({
          code: 'Debug.WriteLine("fake log");',
        }),
      })
    );

    expect(myClass.name).toBe("MyClassName");
    expect(myClass.toString()).toMatchSnapshot();
  });

  it("string codeblock with writer", () => {
    const myClass = csharp.class_({
      name: "MyClassName",
      namespace: "Amplication",
      partial: true,
      access: "public",
      sealed: true,
    });

    myClass.addMethod(
      csharp.method({
        access: "public",
        name: "MyVoidMethod",
        isAsync: true,
        parameters: [
          csharp.parameter({
            name: "simpleParameter",
            type: csharp.Types.integer(),
          }),
          csharp.parameter({
            name: "listParameter",
            type: csharp.Types.list(csharp.Types.string()),
          }),
          csharp.parameter({
            name: "optionalParameter",
            type: csharp.Types.optional(csharp.Types.string()),
          }),
        ],
        body: csharp.codeblock({
          code: (writer) => {
            return;
          },
        }),
      })
    );

    expect(myClass.name).toBe("MyClassName");
    expect(myClass.toString()).toMatchSnapshot();
  });
});
