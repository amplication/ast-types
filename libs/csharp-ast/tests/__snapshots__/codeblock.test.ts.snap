// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code block string codeblock with references 1`] = `
"using System.Diagnostics;

namespace Amplication;

public abstract partial class MyClassName
{
    public async Task MyVoidMethod(int simpleParameter, List<string> listParameter, string? optionalParameter) {
        Debug.WriteLine("fake log");}

}
"
`;

exports[`code block string codeblock with writer 1`] = `
"namespace Amplication;

public sealed partial class MyClassName
{
    public async Task MyVoidMethod(int simpleParameter, List<string> listParameter, string? optionalParameter) {
    }

}
"
`;

exports[`code block string codeblock without references 1`] = `
"namespace Amplication;

public abstract partial class MyClassName
{
    public async Task MyVoidMethod(int simpleParameter, List<string> listParameter, string? optionalParameter) {
        Debug.WriteLine("fake log");}

}
"
`;
